File: src/services/database.service.ts
import { callTauri } from "./tauri";

export const databaseService = {
  /**
   * 备份导出底座 SQLite 数据库物理文件到指定位置
   */
  async exportDatabase(targetPath: string): Promise<void> {
    return await callTauri<void>("export_database", { targetPath });
  },

  /**
   * 从外部备份恢复覆盖 SQLite 并触发客户端自愈重启
   */
  async importDatabase(srcPath: string): Promise<void> {
    return await callTauri<void>("import_database", { srcPath });
  },

  /**
   * 一键重置清空本地 SQLite 数据库与上传文件沙箱
   */
  async resetDatabase(): Promise<void> {
    return await callTauri<void>("reset_database");
  },
};

File: src-tauri/src/services/database_service.rs
use crate::infra::db::DbInfra;
use crate::infra::path::PathProvider;
use crate::infra::AppResult;
use tauri::AppHandle;

pub struct DatabaseService {
    db_infra: DbInfra,
    path_provider: PathProvider,
}

impl DatabaseService {
    pub fn new(app_handle: AppHandle, path_provider: PathProvider) -> Self {
        let db_infra = DbInfra::new(app_handle, path_provider.clone());
        Self {
            db_infra,
            path_provider,
        }
    }

    pub fn export_database(&self, target_path: &str) -> AppResult<()> {
        self.db_infra.export_db(target_path)
    }

    pub fn import_database(&self, src_path: &str) -> AppResult<()> {
        self.db_infra.import_db(src_path)
    }

    pub fn reset_database(&self) -> AppResult<()> {
        self.db_infra.reset_db()
    }

    pub fn check_status(&self) -> AppResult<()> {
        self.init_runtime_schema()
    }

    fn init_test_logs_db(&self) -> AppResult<()> {
        let db_path = self.path_provider.get_db_file_path()?;
        crate::infra::db_nav::DbNavInfra::init_test_logs_db(&db_path)
    }

    pub fn init_runtime_schema(&self) -> AppResult<()> {
        self.init_test_logs_db()?;
        let db_path = self.path_provider.get_db_file_path()?;
        crate::infra::ai_generation_schema::init_ai_generation_schema(&db_path)?;
        Ok(())
    }
}
